### PR TITLE
Feat/execute timeout

### PR DESCRIPTION
**Pull request by L-jasmine**

### CODE REVIEW

**Comment by juntao** [comment]
Hello, I am a [code review bot](https://github.com/flows-network/github-pr-summary/) on [flows.network](https://flows.network/).

It could take a few minutes for me to analyze this PR. Relax, grab a cup of coffee and check back later. Thanks!

**Comment by apepkuss** [comment]
@L-jasmine Please fix the clippy issues on `MacOS` and `Windows` to guarantee all CI workflows pass. Thanks!

File: src/externals/function.rs [hunk 1]
```diff
@@ -243,6 +243,30 @@ impl Func {
...
+    /// * `args` - The arguments passed to the host function.
+    ///
+    /// * `timeout_sec` - The maximum execution time in seconds for the function instance.
+    ///
+    /// # Error
+    ///
+    /// If fail to run the host function, then an error is returned.
+    #[cfg(target_os = "linux")]
+    #[cfg_attr(docsrs, doc(cfg(target_os = "linux")))]
+    pub fn run_timeout(
```
Comment (apepkuss) [inline src/externals/function.rs:261]: Rename `run_timeout` to `run_with_timeout`. Thanks!

File: src/externals/function.rs [hunk 2]
```diff
@@ -264,6 +288,33 @@ impl Func {
...
+    /// * `args` - The arguments passed to the host function.
+    ///
+    /// * `timeout_sec` - The maximum execution time in seconds for the function instance.
+    ///
+    /// # Error
+    ///
+    /// If fail to run the host function, then an error is returned.
+    #[cfg(all(feature = "async", target_os = "linux"))]
+    #[cfg_attr(docsrs, doc(cfg(all(feature = "async", target_os = "linux"))))]
+    pub async fn run_async_timeout(
```
Comment (apepkuss) [inline src/externals/function.rs:307]: Rename `run_async_timeout` to `run_async_with_timeout`. Thanks!

File: src/executor.rs [hunk 3]
```diff
@@ -82,6 +111,33 @@ impl Executor {
             .await
     }
 
+    /// Asynchronously runs a host function instance and returns the results.
```
Comment (apepkuss) [inline src/executor.rs:114]: Refine the comments: `Asynchronously runs a host function instance with a timeout setting.`

Thanks!

File: src/externals/function.rs [hunk 4]
```diff
@@ -243,6 +243,30 @@ impl Func {
         executor.run_func(self, args)
     }
 
+    /// Runs this host function and returns the result.
```
Comment (apepkuss) [inline src/externals/function.rs:246]: Refine the comments: `Runs this host function with a timeout setting.`

File: src/vm.rs [hunk 5]
```diff
@@ -486,6 +486,51 @@ impl Vm {
         }
     }
 
+    /// Runs an exported wasm function in a (named or active) [module instance](crate::Instance).
```
Comment (apepkuss) [inline src/vm.rs:489]: Refine the comments: `Runs an exported wasm function in a (named or active) [module instance](crate::Instance) with a timeout setting.`

Thanks!

File: src/vm.rs [hunk 6]
```diff
@@ -534,6 +579,57 @@ impl Vm {
         }
     }
 
+    /// Asynchronously runs an exported wasm function in a (named or active) [module instance](crate::Instance).
```
Comment (apepkuss) [inline src/vm.rs:582]: Refine the comments: `Asynchronously runs an exported wasm function in a (named or active) [module instance](crate::Instance) with a timeout setting.`

Thanks!

File: src/executor.rs [hunk 7]
```diff
@@ -58,6 +58,35 @@ impl Executor {
...
 
+    /// Runs a host function instance and returns the results.
+    ///
+    /// # Arguments
+    ///
+    /// * `func` - The function instance to run.
+    ///
+    /// * `params` - The arguments to pass to the function.
+    ///
+    /// * `timeout_sec` - The maximum execution time in seconds for the function instance.
```
Comment (apepkuss) [inline src/executor.rs:69]: 1. Rename timeout_sec to timeout.
2. Refine the comments: The maximum execution time (in seconds) of the function to be run.

Thanks!

**Review by apepkuss — CHANGES_REQUESTED** [review]

**Review by apepkuss — CHANGES_REQUESTED** [review]

**Comment by apepkuss** [comment]
@L-jasmine Is it possible to add some unit tests for the new APIs? Thanks!

**Review by apepkuss — APPROVED** [review]

**Pull request merged**